packages/lab/src/slider/internal/useMouseDownTrack.ts

```diff
@@ -0,0 +1,21 @@
+import { RefObject } from "react";
+import { getValue } from "./utils";
+import { SliderChangeHandler } from "../types";
+
+export function useMouseDownTrack(
+  trackRef: RefObject<HTMLDivElement>,
+  min: number,
+  max: number,
+  step: number,
+  setValue: SliderChangeHandler,
+  onChange: SliderChangeHandler | undefined
+) {
+  return {
+    trackProps: {
+      onMouseDown(event: React.MouseEvent) {
+        //@ts-ignore
```

Comment:
struggling to type this, the React event attached to the JSX element doesn't seem to be compatible with the global document.addEventListener(mouse)